src/crypto/chacha20.cpp

```diff
@@ -100,6 +102,10 @@ void ChaCha20::Output(unsigned char* c, size_t bytes)
 
     for (;;) {
         if (bytes < 64) {
+            if (m != nullptr) {
+                for (i = 0;i < bytes;++i) tmp[i] = m[i];
```

Comment:
Just `memcpy(tmp, m, bytes)` might be simpler and easy to read than this for loop.

Reply:
I'm not to familiar with compiler optimisations and stuff but I just checked the reference implementation by DJB and some other C based ChaCha implementations and all of them seems to use the byte per byte assignment. I prefer to leave it as it is.